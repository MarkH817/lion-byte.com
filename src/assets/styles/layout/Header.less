.header {
  align-items: center;
  display: flex;
  font-size: var(--font-size-6);
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 0.5em 0 1em 0;
  @media screen and (max-width: 60rem) {
    flex-direction: column;
    justify-content: center;
  }

  @media screen and (max-width: 40rem) {
    font-size: var(--font-size-5);
  }

  @media screen and (max-height: 40rem) {
    font-size: var(--font-size-5);
  }
}

.header__name {
  flex: 2 1 0;
  font-family: var(--font-nunito);
  font-weight: normal;
  letter-spacing: 0.05em;
}

.header__name-link {
  display: inline-block;
  padding: 0 0.75ch;
  text-decoration: none;
}

.header__nav {
  font-family: var(--font-nunito);
}

.header__nav-list {
  font-size: inherit;

  li {
    display: inline-block;
    margin: 0 var(--space-size-4);
  }
}

.header__nav-link {
  font-weight: normal;
  padding: 0 0.75ch;
  text-decoration: none;
  display: block;
  border-bottom: 0.125rem solid transparent;

  &[aria-current] {
    text-decoration: none;
    border-bottom-color: currentColor;
  }
}
