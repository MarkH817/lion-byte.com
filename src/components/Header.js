import React from 'react'
import { Link } from 'gatsby'
import styled from 'styled-components'

export const HeaderStyles = styled.header`
  align-items: center;
  display: flex;
  font-size: 2em;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 0.5em 0 1em 0;

  @media screen and (max-width: 55em) {
    flex-direction: column;
    justify-content: center;
  }

  /**
   * Ensure that the header is shrunken for screens
   * less than 650 x 650 pixels in dimensions
   */
  @media screen and (max-width: 650px) {
    font-size: 1.5em;
  }

  @media screen and (max-height: 650px) {
    font-size: 1.5em;
  }

  .name {
    flex: 2 1 0;
    font-family: ${props => props.theme.headerFont};
    font-weight: bold;

    a {
      display: inline-block;
      padding: 0 0.75ch;
      text-decoration: none;
    }
  }

  nav {
    font-family: ${props => props.theme.headerFont};
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-around;

    a {
      font-weight: bold;
      display: inline-block;
      padding: 0 0.75ch;
      text-align: center;
      text-decoration: none;
    }
  }
`

export function Header () {
  return (
    <HeaderStyles>
      <div className='name'>
        <Link to='/'>Mark Hernandez (lion-byte)</Link>
      </div>

      <nav>
        <Link to='/'>Home</Link>
        <Link to='/about'>About</Link>
        <Link to='/blog'>Blog</Link>
      </nav>
    </HeaderStyles>
  )
}

export default Header
