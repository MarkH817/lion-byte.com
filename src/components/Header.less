.header {
  align-items: center;
  display: flex;
  font-size: @font-size-6;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 0.5em 0 1em 0;
  @media screen and (max-width: @screen-lg) {
    flex-direction: column;
    justify-content: center;
  }

  @media screen and (max-width: @screen-sm) {
    font-size: @font-size-5;
  }

  @media screen and (max-height: @screen-sm) {
    font-size: @font-size-5;
  }
}

.header__name {
  flex: 2 1 0;
  font-family: @font-header;
  font-weight: normal;
  letter-spacing: 0.05em;
}

.header__name-link {
  display: inline-block;
  padding: 0 0.75ch;
  text-decoration: none;
}

.header__nav {
  font-family: @font-header;
}

.header__nav-list {
  font-size: 1em;
}

.header__nav-link {
  font-weight: normal;
  padding: 0 0.75ch;
  text-decoration: none;

  &[aria-current] {
    text-decoration: none;
    border-bottom: 0.1em solid currentColor;
  }
}
