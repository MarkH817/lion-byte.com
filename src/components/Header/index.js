import React from 'react'
import Link from 'gatsby-link'

import Navigation from '../Navigation'

const Header = () => (
  <header>
    <div>
      <h1>
        <Link to='/'>Mark Hernandez</Link>
      </h1>
    </div>

    <Navigation />
  </header>
)

export default Header
