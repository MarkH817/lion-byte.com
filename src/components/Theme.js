import React from 'react'
import { ThemeProvider, createGlobalStyle } from 'styled-components'
import '@reach/skip-nav/styles.css'
import 'prismjs/themes/prism-okaidia.css'
import 'typeface-nunito'
import 'typeface-open-sans'

const theme = {
  headerFont: `'Nunito', Arial, Helvetica, sans-serif`,
  textFont: `'Open Sans', Arial, Helvetica, sans-serif`,
  blue: '#28579d',
  red: '#e73212',
  black: '#2e2e31',
  grey: '#3a3a3a',
  lightGrey: '#e1e1e1',
  offWhite: '#ededed',
  maxWidth: '750px',
  bs: '0 12px 24px 0 rgba(0, 0, 0, 0.09)'
}

const GlobalStyles = createGlobalStyle`
  * {
    box-sizing: border-box;

    &:focus {
      outline: 0.1rem solid ${props => props.theme.red};
    }

    &::selection {
      background-color: ${props => props.theme.red};
      color: #ffffff;
    }
  }

  body {
    font-family: ${props => props.theme.textFont};
    line-height: 1.3;
    margin: 0;
    padding: 0;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${props => props.theme.headerFont};
  }

  a {
    color: ${props => props.theme.blue};

    &:active {
      color: ${props => props.theme.red};
    }
  }

  p {
    margin-bottom: 2em;
  }

  [data-reach-skip-link] {
    color: cornflowerblue;
    text-decoration: none;

    &:focus {
      outline-color: currentColor;
    }
  }

  @keyframes fadein {
    0% {
      opacity: 0;
    }

    100% {
      opacity: 1;
    }
  }
`

export const Theme = props => {
  const { children } = props

  return (
    <ThemeProvider theme={theme}>
      <React.Fragment>
        <GlobalStyles />

        {children}
      </React.Fragment>
    </ThemeProvider>
  )
}

export default Theme
