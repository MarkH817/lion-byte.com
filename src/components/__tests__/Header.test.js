import React from 'react'
import { render } from '@testing-library/react'

import Theme from '../Theme'
import Header from '../Header'

describe('Nav', () => {
  test('renders correctly', () => {
    const { container } = render(
      <Theme>
        <Header />
      </Theme>
    )

    expect(container).toMatchSnapshot()
  })
})
