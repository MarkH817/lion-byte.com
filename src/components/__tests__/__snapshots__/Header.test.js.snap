// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="header__name"
    >
      <a
        class="header__name-link"
        href="/"
      >
        Mark Hernandez
      </a>
    </div>
    <nav
      aria-labelledby="main-nav-label"
      class="header__nav"
    >
      <span
        class="screen-reader"
        id="main-nav-label"
      >
        Main
      </span>
      <ul
        class="header__nav-list"
      >
        <li>
          <a
            class="header__nav-link"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="header__nav-link"
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            class="header__nav-link"
            href="/blog"
          >
            Blog
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
